service/src/main/java/uk/gov/hmcts/reform/fpl/controllers/ActionCaseManagementOrderController.java

```diff
@@ -76,16 +82,29 @@ public AboutToStartOrSubmitCallbackResponse handleMidEvent(
 
         CaseDetails caseDetails = callbackRequest.getCaseDetails();
         CaseData caseData = mapper.convertValue(caseDetails.getData(), CaseData.class);
+        CaseManagementOrder order = caseData.getCmoToAction();
+
+        String nextHearingId = (String) caseDetails.getData().get("nextHearingDateList");
```

Comment:
`nextHearingDateList` unexpectedly comes in as UUID as string rather than a Dynamiclist on the mid event. Because of this had to make some changes (including casting) to get hearing booking via this ID.

Reply:
Resolved